geoh5py/workspace/workspace.py

```diff
@@ -347,24 +347,26 @@ def create_from_concatenation(self, attributes):
         if "Name" in attributes:
             attributes["Name"] = attributes["Name"].replace("\u2044", "/")
 
+        recovered_entity = None
+
         if "Object Type ID" in attributes:
             recovered_entity = self.create_entity(
                 ObjectBase,
                 save_on_creation=False,
                 **{
                     "entity": attributes,
-                    "entity_type": {"uid": attributes["Object Type ID"]},
+                    "entity_type": {"uid": attributes.pop("Object Type ID")},
                 },
             )
 
-        else:
+        elif "Type ID" in attributes:
             recovered_entity = self.create_entity(
                 Data,
                 save_on_creation=False,
                 **{
                     "entity": attributes,
                     "entity_type": self.fetch_type(
-                        uuid.UUID(attributes["Type ID"]), "Data"
+                        uuid.UUID(attributes.pop("Type ID")), "Data"
                     ),
                 },
             )
```

Comment:
else? 
No error?

Reply:
It added "Type ID" on the class attributes, which not great, but would also make the `compare_entities` to crash in tests

Reply:
perfect, mark this as resolved